lib/sqlalchemy/engine/url.py

```diff
@@ -34,6 +34,7 @@
 from urllib.parse import parse_qsl
 from urllib.parse import quote_plus
 from urllib.parse import unquote
+from unittest.mock import Mock
```

Comment:
I'm -1 on importing unittest code here

if we need to guard against this it's probably best to go with `hasattr(name_or_url, '_instantiate_plugins')` or something along these lines